lua/renamer/utils.lua

```diff
@@ -64,8 +64,8 @@ function utils.set_qf_list(changes)
                 qf_list[i] = {
                     text = line and line[1],
                     filename = file,
-                    lnum = row,
-                    col = col,
+                    lnum = row + 1,
+                    col = col + 1,
```

Comment:
For LSP responses with the `changes` field, it needs 0 indexing. Try with `rust-analyzer` for example or the Lua LSP server. At least that is how it works for me

Reply:
I've tested it for `documentChanges` and `changes` as well. This change is for the qflist (which does 1 index), and doesn't affect any LSP server responses. Here's it working after the change, with `rust_analyzer`:
![image](https://user-images.githubusercontent.com/62098008/142859398-f8c6acc4-991c-43f8-b5a4-4b7995cf42f0.png)
Perhaps the tests are incorrect :))

Reply:
Try doing the same test, but renaming something on the first line of the file. That's where I saw the issue

Reply:
Something like this?
https://user-images.githubusercontent.com/62098008/142869957-ce96365d-4d12-4b85-8264-ef244b206a47.mp4

Reply:
Yep, that seems about right. Wonder why for Lua I needed to start from 0 :thinking: you wouldn't happen to also have the Lua LSP server right? I know for a fact that one has the `changes` field only. Otherwise, I guess we should fix the tests and we can merge it.

I should also write better tests for the `buf_request` calls that are happening. Clearly a lot can get lost there.

Reply:
I mean the changes I made are LSP agnostic as they just ensure that qflist displays them properly (it previously wasn't - this is a separate issue). Here is it working with sumneko_lua:

https://user-images.githubusercontent.com/62098008/142891207-054c57ae-6acf-4f7f-a9cb-1dd463a830ba.mp4

Reply:
Yep looks good. Seems like it was only a problem on my end. Not sure what caused it.